sdk/communication/Azure.Communication.Common/src/EntraTokenCredential.cs

```diff
@@ -115,6 +116,37 @@ private HttpMessage CreateRequestMessage()
             return message;
         }
 
+        private RequestUriBuilder CreateRequestUri()
+        {
+            var uri = new RequestUriBuilder();
+            uri.Reset(new Uri(_resourceEndpoint));
+
+            var (endpoint, apiVersion) = DetermineEndpointAndApiVersion();
+            uri.AppendPath(endpoint, false);
+            uri.AppendQuery("api-version", apiVersion, true);
+            return uri;
+        }
+
+        private (string Endpoint, string ApiVersion) DetermineEndpointAndApiVersion()
+        {
+            if (_scopes == null || !_scopes.Any())
+            {
+                throw new ArgumentException($"Scopes validation failed. Ensure all scopes start with either {TeamsExtentionsScopePrefix} or {EntraScopePrefix}.", nameof(_scopes));
```

Comment:
as above, I think it's convenient to have the communication/clients .default scope as the default when no scopes are set

Reply:
I think it's better to have these checks, as the developers might provide invalid scopes

Reply:
We've discussed more and decided to keep Scopes optional to have more frictionless experience for the communication/clients scenario. For TeamsPhone scenarios, developers will have to explicitly set the scopes.